### PR TITLE
AWS: Fix flaky TestS3RestSigner

### LINKED ISSUE #10599: `TestS3RestSigner` is flaky
### Apache Iceberg version

main (development)

### Query engine

None

### Please describe the bug 🐞

The 

```
TestS3RestSigner > executionError FAILED
    org.opentest4j.AssertionFailedError: [should not have any expired token that required a refresh] 
    expected: 0L
     but was: 1L
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at org.apache.iceberg.aws.s3.signer.TestS3RestSigner.afterClass(TestS3RestSigner.java:120)
```

/cc @nastra

### PR DESCRIPTION
**Pull request by nastra**
fixes #10599

### CODE REVIEW

File: aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java [hunk 1]
```diff
@@ -107,7 +107,13 @@ public static void afterClass() throws Exception {
 
     ScheduledThreadPoolExecutor executor =
         ((ScheduledThreadPoolExecutor) validatingSigner.icebergSigner.tokenRefreshExecutor());
-    // token expiration is set to 100s so there should be exactly one token scheduled for refresh
+    // token expiration is set to 10000s so there should be exactly one token scheduled for refresh.
```
Comment (amogh-jahagirdar) [inline aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java:110]: Could we link or somehow reference the S3SignerServlet class in this comment? That's where it's actually set.

**Review by amogh-jahagirdar — APPROVED** [review]
Overall LGTM, just had a comment on clarifying the code comment a bit more. 
It'd be ideal to be able to not have to rely on timing but I also want to unblock flaky tests that others may be hitting on the PR and since this is a test we can always change it later if we are able to get a better way of testing what we want to test.

File: aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java (hunk 1 above)
Comment (nastra) [reply aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java]: sure, I updated the comment to mention that it's set in the servlet

**Comment by nastra** [comment]
> It'd be ideal to be able to not have to rely on timing but I also want to unblock flaky tests that others may be hitting on the PR and since this is a test we can always change it later if we are able to get a better way of testing what we want to test.


Yes I agree, we don't want to rely on the timing of the test class itself and typically running this test locally took < 10 seconds so I initially just picked 100 seconds for the token expiration, which on the other hand didn't seem enough for CI.
In the end, all we care about in this particular scenario is that there's only a single token left in the refresh scheduler's queue in order to make sure #7270 doesn't happen

**Pull request merged**